pkg/apiserver/apiserver_test.go

```diff
@@ -785,6 +796,105 @@ func TestDelete(t *testing.T) {
 	}
 }
 
+func TestDeleteWithOptions(t *testing.T) {
+	storage := map[string]RESTStorage{}
+	simpleStorage := SimpleRESTStorage{}
+	ID := "id"
+	storage["simple"] = &simpleStorage
+	handler := handle(storage)
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	grace := int64(300)
+	item := &api.DeleteOptions{
+		GracePeriod: &grace,
+	}
+	body, err := codec.Encode(item)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	client := http.Client{}
+	request, err := http.NewRequest("DELETE", server.URL+"/api/version/simple/"+ID, bytes.NewReader(body))
```

Comment:
Are there intermediate proxies that drop request body on HTTP DELETE operations?  Do all common HTTP libraries allow you to send a request entity on a DELETE?  I am wondering why this approach instead of using a header X-Grace-Period or something similar?